widget/view.controller.js

```diff
@@ -72,19 +77,55 @@
             scheduleFrequency: null
         };
         $scope.outbreakAlertSeverityList = ['Critical', 'High', 'Medium'];
-        $scope.$watch('activeTab', function ($newTab, $oldTab) {
-            if (!$oldTab) {
-                // skip first run
-                return;
-            }
-            $state.go('.', {
-                tab: $scope.selectedEnv.huntTools[0].itemValue
-            }, {
-                notify: false,
-                location: 'replace'
+
+        $scope.parent_wf_id = '';
+        var subscription;
+        initWebsocket();
+
+        $scope.$on('websocket:reconnect', function () {
+            initWebsocket();
+        });
+
+        function initWebsocket() {
```

Comment:
As discussed,
1. You don't need to subscribe at the start of the widget, make it when its needed
2. you don't need to check for all executed PBs
3. Use task id of your executed PB and check for that only.